src/robusta/core/sinks/slack/slack_sink_params.py

```diff
@@ -35,19 +33,43 @@ def validate_channel_override(cls, v: str):
                 )
                 raise ValueError(err_msg)
         return v
+    
+    def normalize_placeholder(cls, s):
+        def repl(match):
+            return cls.normalize_key_string(match.group(0))
+        
+        result = s.replace(ANNOTATIONS_PREF, '').replace(LABELS_PREF, '')
+        # Use regular expression to find and replace inside "${}"
+        result = regex.sub(r'\$\{[^}]+\}', repl, result)
+        result = regex.sub(rf'\$({ANNOTATIONS_PREF})|({LABELS_PREF})', repl, result)
+        return result
+    
+    def normalize_key_string(cls, s: str) -> str:
+        return s.replace('/', '_').replace('.', '_').replace('-', '_')
+    
+    def normalize_dict_keys(cls, metadata: Dict) -> Dict:
+        return {cls.normalize_key_string(k):v for k, v in metadata.items()}
 
     def get_slack_channel(self, cluster_name: str, labels: Dict, annotations: Dict) -> str:
         if self.channel_override:
+            channel = self.channel_override
+            
+            # Update labels with CLUSTER_PREF and create annotations with defaults
             labels.update({CLUSTER_PREF: cluster_name})
-            channel = self.channel_override.replace(LABELS_PREF, "")
-            channel = Template(channel).safe_substitute(labels)
-            channel = channel.replace(ANNOTATIONS_PREF, "")
             annots = defaultdict(lambda: MISSING)
             annots.update(annotations)
-            channel = Template(channel).safe_substitute(annots)
+
+            # Normalize channel and labels/annotations keys
+            channel_normalized = self.normalize_placeholder(channel)
+            labels_annotaions_normalized = self.normalize_dict_keys(labels | annots)
```

Comment:
I think the code here miss the edge case, where there's some value in both annotations and labels.
For example:
If you configure:
`channel_override: "labels.channel`

And you have both a `label` channel, and an `annotation` channel

Might be better to call it separately for `labels` and `annotations` to handle this potential edge case

Reply:
OK got it.